From 461456c7f43894e50bb53f345cea248d8961c7f8 Mon Sep 17 00:00:00 2001
From: SpiritCroc <spiritcroc@gmail.com>
Date: Sun, 4 Jun 2017 20:25:56 +0200
Subject: [PATCH 038/296] Fix empty recents view in multi window mode

An empty recents view will be launched if recents become
visible and another visible task is the resumed (top) task.

Fix this by reloading tasks in onStart.
---
 .../SystemUI/src/com/android/systemui/recents/RecentsActivity.java    | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/recents/RecentsActivity.java b/packages/SystemUI/src/com/android/systemui/recents/RecentsActivity.java
index 86b77900fdb..836c2f9f432 100644
--- a/packages/SystemUI/src/com/android/systemui/recents/RecentsActivity.java
+++ b/packages/SystemUI/src/com/android/systemui/recents/RecentsActivity.java
@@ -411,6 +411,10 @@ public class RecentsActivity extends Activity implements ViewTreeObserver.OnPreD
         // Notify of the next draw
         mRecentsView.getViewTreeObserver().addOnPreDrawListener(mRecentsDrawnEventListener);
 
+        if (isInMultiWindowMode()) {
+             reloadTaskStack(true /* isInMultiWindowMode */, false /* sendConfigChangedEvent */);
+        }
+
         // If Recents was restarted, then it should complete the enter animation with partially
         // reset launch state with dock, app and home set to false
         Object isRelaunching = getLastNonConfigurationInstance();
-- 
2.11.0

